container/image.bzl

```diff
@@ -216,9 +161,25 @@ def _repository_name(ctx):
   # the v2 registry specification.
   return _join_path(ctx.attr.repository, ctx.label.package)
 
+def _evalute_merge_env(env, new_env):
+  """Expand new environment variables properly and merge them.
+  Example:
+  - input: env={'foo':'bar', 'PATH':'$PATH:a:b'}
+           new_env={'X':'Y', 'PATH':'$PATH:c' }
+  - output: env={'foo':'bar', 'X':'Y', 'PATH':'$PATH:a:b:c'}
+  """
+  for k, v in new_env.items():
+    elems = []
+    for e in v.split(":"):
+      if e.startswith("$") and e[1:] in env.keys():
```

Comment:
This doesn't handle `a:$PATH:b` or `${PATH}:a:b`.

I'd really rather see this go away in favor of `_image_config`.

Reply:
`PATH` is also not the environment variable that supports read-modify-write, so splitting on `:` is also insufficient.

Reply:
Done. 

Now use //container:create_image_config to generate config layer by layer so the envvars can be handled correctly.